### ISSUE TITLE
model performance

### ISSUE THREAD

**Issue by jhoh10** [airalcorn2/Recurrent-Convolutional-Neural-Network-Text-Classifier#12]
I tried to recreate the model performance of the this network on the **20NewsGroups** dataset. I used google's pretrained word2vec embedding with vector dimensionality of 300, which was the only difference to the paper outside of using LSTMs vs the paper's BD RNN's.

During the training, I tracked the validation set auc_roc.

epoch 1: roc-auc on val - 0.840
epoch 2: roc-auc on val - 0.995
epoch 3: roc-auc on val - 0.996
epoch 4: roc-auc on val - 0.998
epoch 5: roc-auc on val - 0.997
epoch 6: roc-auc on val - 0.998
epoch 7: roc-auc on val - 0.997

At model test time, after 7 epochs, my f1 macro score was 0.91, compared to 0.9649 reported by the paper.

Have you run into issues with the model matching the paper? If so, any ideas what could be causing the discrepancy? And do you think I am training this network for too many epochs?

Much appreciated!

**Comment by airalcorn2** [comment]
The word vectors could certainly make a big difference, especially since word2vec has a dimensionality of 300 and the word embeddings used in the paper only have a dimensionality of 50. Because the hidden layer used in the paper has 100 units, that's a difference of 100 * 300 - 100 * 50 = 25,000 parameters in the vanilla RNN model, which is a lot when considering the data set is already prone to overfitting because it's so small. You also have to consider the fact that the error landscape is probably riddled with local optima, so it's extremely unlikely you would converge on the parameters found in the paper.
